src/EFTools/EntityDesignerVersioningFacade/ReverseEngineerDb/SchemaDiscovery/EntityStoreSchemaGeneratorDatabaseSchemaLoader.cs

```diff
@@ -487,5 +527,46 @@ ORDER BY
             ,   sp.Name
             ,   sp.Ordinal
             ";
+
+        /// <summary>
+        /// This interceptor is added to work around a problem with SQL Server
+        /// cardinality estimation: https://github.com/aspnet/EntityFramework6/issues/4
+        /// which in turn causes bad performance running the queries above on large models.
+        /// We can workaround this by directing SQL Server to use MERGE JOINs.
+        /// </summary>
+        private sealed class AddOptionMergeJoinInterceptor : IDbCommandInterceptor
```

Comment:
I think there is probably an abstract base class that could be used here instead of using the interface directly. Can you check?

Reply:
There is. Fixed in next commit.